sentry_streams/sentry_streams/adapters/arroyo/msg_wrapper.py

```diff
@@ -0,0 +1,51 @@
+import time
+from typing import Optional, TypeVar, Union, cast
+
+from arroyo.processing.strategies.abstract import ProcessingStrategy
+from arroyo.types import FilteredPayload, Message, Value
+
+from sentry_streams.adapters.arroyo.routes import Route, RoutedValue
+from sentry_streams.pipeline.message import Message as StreamsMessage
+
+TPayload = TypeVar("TPayload")
+
+
+class MessageWrapper(ProcessingStrategy[Union[FilteredPayload, TPayload]]):
+    """
+    Custom processing strategy which either produces an incoming message via a given Producer
+    if the Route of the message matches this strategy's Route, or forwards the message
+    to the next strategy provided.
+    """
+
+    def __init__(
+        self,
+        route: Route,
+        next_step: ProcessingStrategy[Union[FilteredPayload, RoutedValue]],
+    ) -> None:
+        self.__next_step = next_step
+        self.__route = route
+
+    def submit(self, message: Message[Union[FilteredPayload, TPayload]]) -> None:
+        now = time.time()
+        if not isinstance(message.payload, FilteredPayload):
+            msg = StreamsMessage(message.payload, [], now, None)
+
+            routed_msg: Message[RoutedValue] = Message(
+                Value(committable=message.value.committable, payload=RoutedValue(self.__route, msg))
+            )
```

Comment:
This is a good sign that we will not stick with arroyo in the long run:
StreamsMessage is the payload of RoutedMessage 
which  is the payload of the value of Message. 
That's a lot

Reply:
Yeah unfortunately had to alias our top-level API's `Message` as a `StreamsMessage` in order to not clash with Arroyo's `Message`

Reply:
I had kind of hoped for a solution where all of it can be managed at the top-level API code, but given that we want to expose _some_ metadata to the user, like processing time, the `StreamsMessage` type has to be managed in the Arroyo steps as well